test/HttpRequestTests.cpp

```diff
@@ -307,7 +310,11 @@ void HttpRequestTests::testSimpleGet()
 
         std::unique_lock<std::mutex> lock(mutex);
 
-        LOK_ASSERT(httpSession->asyncRequest(httpRequest, pollThread));
+        httpSession->setConnectFailHandler([this]() {
+            LOK_ASSERT(false);
```

Comment:
`LOK_ASSERT_FAIL("Unexpected connection failure");` might be more helpful (with a similar message).

Reply:
will do